### PR TITLE
Partially address graph blowup in index based merges

### PR DESCRIPTION
**Pull request by phofl**
cc @fjetter

### CODE REVIEW

**Review by hendrikmakait — APPROVED** [review]
Thanks, @phofl! Looks reasonable to me

File: dask_expr/_shuffle.py [hunk 1]
```diff
@@ -241,9 +241,7 @@ def _lower(self):
                 ]
             drop_columns = partitioning_index.copy()
         elif index_shuffle:
-            frame = Assign(frame, "_partitions_0", frame.index)
-            partitioning_index = ["_partitions_0"]
-            drop_columns = partitioning_index.copy()
+            dtypes = np.float64 if _is_numeric_cast_type(dtype) else None
```
Comment (hendrikmakait) [inline dask_expr/_shuffle.py:244]: dtype isn't defined here.

**Review by hendrikmakait — CHANGES_REQUESTED** [review]

File: dask_expr/_shuffle.py (hunk 1 above)
Comment (phofl) [reply dask_expr/_shuffle.py]: Thx, I should stop making non-trivial changes in the ui

**Review by hendrikmakait — APPROVED** [review]
Thanks for fixing!

**Pull request merged**